### PR TITLE
Create iframe within amp-ads for inabox

### PR DESCRIPTION
**Pull request by zhaoqin**
Create iframe within amp-ads for inabox

- create iframe within the amp-ads if inabox
- add tests

### CODE REVIEW

File: extensions/amp-analytics/0.1/amp-analytics.js [hunk 1]
```diff
@@ -333,7 +333,9 @@ export class AmpAnalytics extends AMP.BaseElement {
         `${TAG}: No friendly parent amp-ad element was found for ` +
         'amp-analytics tag with iframe transport.');
 
-    this.iframeTransport_ = new IframeTransport(this.getAmpDoc().win,
+    this.iframeTransport_ = new IframeTransport(
+        // We create a nested corss-domain iframe within the ads if isInabox.
```
Comment (jonkeller) [inline extensions/amp-analytics/0.1/amp-analytics.js]: Nit: "cross"

File: extensions/amp-analytics/0.1/test/test-amp-analytics.js [hunk 2]
```diff
@@ -1773,4 +1773,60 @@ describes.realWin('amp-analytics', {
       expect(getAnalyticsTag(getConfig()).getPriority()).to.equal(1);
     });
   });
+
+  describe('inabox nested transport iframe', () => {
+    let ampMode;
+    beforeEach(() => {
+      ampMode = env.win.AMP_MODE
```
Comment (jonkeller) [inline extensions/amp-analytics/0.1/test/test-amp-analytics.js]: Nit: missing semicolon. Maybe also change variable name to indicate it's saving the old value.

File: extensions/amp-analytics/0.1/amp-analytics.js (hunk 1 above)
Comment (zhaoqin) [reply extensions/amp-analytics/0.1/amp-analytics.js]: done

File: extensions/amp-analytics/0.1/test/test-amp-analytics.js (hunk 2 above)
Comment (zhaoqin) [reply extensions/amp-analytics/0.1/test/test-amp-analytics.js]: done

**Comment by jonkeller** [comment]
Travis failed, in different places in the two builds, so I restarted it in hopes that it's just flakiness.

File: extensions/amp-analytics/0.1/amp-analytics.js [hunk 3]
```diff
@@ -333,7 +333,9 @@ export class AmpAnalytics extends AMP.BaseElement {
         `${TAG}: No friendly parent amp-ad element was found for ` +
         'amp-analytics tag with iframe transport.');
 
-    this.iframeTransport_ = new IframeTransport(this.getAmpDoc().win,
+    this.iframeTransport_ = new IframeTransport(
+        // We create a nested cross-domain iframe within the ads if isInabox.
```
Comment (keithwrightbos) [inline extensions/amp-analytics/0.1/amp-analytics.js]: Nit, change comment to: // Create 3p transport frame within creative frame if inabox

File: extensions/amp-analytics/0.1/test/test-amp-analytics.js [hunk 4]
```diff
@@ -1773,4 +1773,60 @@ describes.realWin('amp-analytics', {
...
+
+    afterEach(() => {
+      doc.body.classList.remove('i-amphtml-element');
+      env.win.AMP_MODE = origAmpMode;
+    });
+
+    it('sends a basic hit', function() {
+      const analytics = getAnalyticsTag(trivialConfig);
+      return waitForSendRequest(analytics).then(() => {
+        expect(sendRequestSpy.withArgs('https://example.com/bar').calledOnce)
```
Comment (keithwrightbos) [inline extensions/amp-analytics/0.1/test/test-amp-analytics.js]: expect(sendRequestSpy...).to.be.calledOnce.

File: extensions/amp-analytics/0.1/amp-analytics.js [hunk 5]
```diff
@@ -333,7 +333,9 @@ export class AmpAnalytics extends AMP.BaseElement {
         `${TAG}: No friendly parent amp-ad element was found for ` +
         'amp-analytics tag with iframe transport.');
 
-    this.iframeTransport_ = new IframeTransport(this.getAmpDoc().win,
+    this.iframeTransport_ = new IframeTransport(
```
Comment (lannka) [inline extensions/amp-analytics/0.1/amp-analytics.js]: @zhaoqin I'm wondering why this change was needed. in the case of inabox, `this.win` is always identical to `this.getAmpDoc().win`.

File: extensions/amp-analytics/0.1/test/test-amp-analytics.js [hunk 6]
```diff
@@ -1773,4 +1773,60 @@ describes.realWin('amp-analytics', {
       expect(getAnalyticsTag(getConfig()).getPriority()).to.equal(1);
     });
   });
+
+  describe('inabox nested transport iframe', () => {
```
Comment (lannka) [inline extensions/amp-analytics/0.1/test/test-amp-analytics.js]: the test here is unrelevant to the above change.
